python/benchmark/gen_data_distributed.py

```diff
@@ -580,6 +592,12 @@ def make_regression_udf(iter: Iterable[pd.DataFrame]) -> Iterable[pd.DataFrame]:
 class SparseRegressionDataGen(DataGenBaseMeta):
     """Generate sparse regression dataset using a distributed version of sklearn.datasets.regression,
     including features and labels.
+
+    SparseRegressionDataGen is a deterministic data generator when running in Spark local mode. It produces
+    identical DataFrames when initialized with the same random state. However, in Spark standalone mode, it may occasionally
+    yield different DataFrames even with the same random state. This is because each partition derives its seed from
+    a combination of the provided random state and the partition index. Since partition indices are not guaranteed to be continuous
+    or consistent across runs in standalone mode, this can lead to non-deterministic output despite a fixed random state.
```

Comment:
Should be consistent as long as number of partitions is the same but that could change depending on number of executor-cores for example.    Maybe need to give option to set the num partitions argument to spark.range , the base dataframe/rdd.

Reply:
Added a warning message when output_num_files not specified.